devices/At24CXX/AT24Base.cs

```diff
@@ -0,0 +1,207 @@
+        /// Reads multiple bytes from the device starting from the given address.
+        /// </summary>
+        /// <param name="address">The address to read from.</param>
+        /// <param name="length">The number of bytes to read.</param>
+        /// <returns>The bytes read from the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public byte[] Read(int address, int length)
+        {
+            VerifyMemoryAddress(address, Size);
+
+            byte[] readBuffer = new byte[length];
+            byte[] writeBuffer = CreateWriteBuffer(address, 0);
+
+            _i2cDevice.WriteRead(writeBuffer, readBuffer);
+
+            return readBuffer;
+        }
+
+        /// <summary>
+        /// Writes a single byte to the device at the given address.
+        /// </summary>
+        /// <param name="address">The address to write to.</param>
+        /// <param name="value">The byte to be written into the device.</param>
+        /// <returns>The number of bytes successfully written to the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public uint WriteByte(int address, byte value)
+        {
+            VerifyMemoryAddress(address, Size);
+
+            byte[] writeBuffer = CreateWriteBuffer(address, 1);
+            writeBuffer[2] = value;
+
+            I2cTransferResult result = _i2cDevice.Write(writeBuffer);
+
+            // Account for memory address that was sent to device along with the data to be written
+            return result.BytesTransferred - 2;
+        }
+
+        /// <summary>
+        /// Writes multiple bytes to the device starting from the given address.
+        /// </summary>
+        /// <param name="address">The address to write to.</param>
+        /// <param name="data">The bytes to be written to the device.</param>
+        /// <returns>The number of bytes successfully written to the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public uint Write(int address, byte[] data)
+        {
+            VerifyMemoryAddress(address, Size);
+
+            if (data.Length > PageSize)
+            {
+                throw new ArgumentOutOfRangeException(nameof(data), "Length of data to be written cannot exceed device page size.");
+            }
+
+            byte[] writeBuffer = CreateWriteBuffer(address, data.Length);
+            data.CopyTo(writeBuffer, 2);
+
+            I2cTransferResult result = _i2cDevice.Write(writeBuffer);
+ 
+            // Account for memory address that was sent to device along with the data to be written
+            return result.BytesTransferred - 2;
+        }
+
+        private void Dispose(bool disposing)
```

Comment:
we don't really use this pattern anymore. You can move all the logic in the Dispose() function